es-app/src/FileData.cpp

```diff
@@ -81,6 +81,37 @@ const std::string& FileData::getThumbnailPath() const
 		return metadata.get("image");
 }
 
+const std::vector<FileData*>& FileData::getChildrenListToDisplay() {
+	
+	FileFilterIndex* idx = mSystem->getIndex();
+	if (idx->isFiltered()) {
+		mFilteredChildren.clear();
+		for(auto it = mChildren.begin(); it != mChildren.end(); it++)
+		{
+			if (idx->showFile((*it))) {
+				mFilteredChildren.push_back(*it);
+			}
+		}
+
+		// need to check if list is empty, and if so add a placeholder node
+		if (mFilteredChildren.size() == 0) 
+		{
+			// add a placeholder
+			// for now, just create a new FileData of type Folder, with no children
+			// maybe in the future makes sense to create a third type of FileData?
+			FileData* placeholder = new FileData(FOLDER, "<No Results Found for Current Filter Criteria>", mSystem);
```

Comment:
I would recommend going ahead and creating a PLACEHOLDER type.  You could then removing the placeholder variable and methods from FileData and check against the type instead.